pkg/machine/os/ostree.go

```diff
@@ -0,0 +1,137 @@
+//go:build amd64 || arm64
+// +build amd64 arm64
+
+package os
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/sirupsen/logrus"
+)
+
+// OSTree deals with operations on ostree based os's
+type OSTree struct { //nolint:revive
+}
+
+// Apply takes an OCI image and does an rpm-ostree rebase on the image
+// If no containers-transport is specified,
+// apply will first check if the image exists locally, then default to pulling.
```

Comment:
I have not looked at the full code but I don't understand why you need pull anything at all in podman or check if the image exists in the store.
AFAICT rpm-ostree will pull the image on its own with ` ostree-unverified-registry` per https://coreos.github.io/rpm-ostree/container/#rebasing-a-client-system

If we just pass the input to rpm-ostree we could make the code a bit simpler.

Reply:
This essentially is a work-around for two things:
1. rpm-ostree requires you to specify the containers-transport when pulling. The pull in podman allows the behavior of os apply to match other podman commands, where you only pull if the image does not exist in storage already.
2. This works around the root/rootless issue. Podman machines are by default set up using a rootless connection. rpm-ostree needs to be run as root. If a user wants to use an image in containers-storage, rpm-ostree will look at the root storage, and not the user storage, which is unexpected behavior. Exporting to an oci-dir works around this, without nagging the user to configure the machine in rootful mode.

Reply:
OK sounds reasonable, maybe add this as comment in the code.

Reply:
Will do, thanks!